SSL Certificate checking

av2k opened the pull request:
Hello,

this is the code we use since about 9 months for certificate checking.

It adds a new column in the database to store the minimum days of remaining certificate validity and uses OpenSSL to get the server certificate details.

Commit:
Add header_name & header_value to the server object

Commit:
Add optionnal header check on StatusUpdater

Commit:
Refactor header checking to be sure that we can't crash this !

gmoigneu committed:
Fix typo on Installer.php

mubix committed:
Use existing mirror

apt.sw.be no longer exists

mubix committed:
Use existing mirror

apt.sw.be no loger exists

AndyGaskell committed:
Adding nexmo sms to the select

Just adding nexmo as an option to the select drop down.

This is a follow up to work done on Issue #170

AndyGaskell committed:
Rename Nexmo.class.php to Nexmo.php

Fix for nexmo sms, sending sms didn't work till I renamed the file.

I'm not sure why Nexmo.class.php and Octopush.class.php have the .class bit in the filename.

This is a follow up to Issue #170 Implement Nexmo SMS

Commit:
preparing for 3.2.0 release

Commit:
Update HistoryGraph.php

with several php-settings there might be problems with timestamp output - so it would be better to use number_format for output. otherwise the graph wont show...

decryptus committed:
Update update.tpl.html

Fixed for attributes pushover_key and pushover_device.

sadortun committed:
Merge pull request #464 from decryptus/patch-1

Update update.tpl.html

sadortun committed:
Merge pull request #435 from AndyGaskell/patch-1

Adding nexmo sms to the select

sadortun committed:
Merge pull request #437 from AndyGaskell/patch-3

Rename Owctopush.class.php to Octopush.php

sadortun committed:
Merge pull request #432 from mubix/patch-1

Use existing mirror

sadortun committed:
Merge pull request #433 from mubix/patch-2

Use existing mirror

sadortun committed:
Merge pull request #343 from gmoigneu/headers

Add optionnal header value check (issue #214)

sadortun committed:
Merge pull request #454 from rabauss/patch-1

Update HistoryGraph.php

vus520 committed:
#465 Japanese language added

sadortun committed:
Merge pull request #475 from nutsteam/nutsteam/lang_jp

#465 Japanese language added

Commit:
Add Certificate checking

Commit:
Merge branch 'develop' of https://github.com/av2k/phpservermon into develop

jimwebb commented:
This patch worked well for me, with two small issues:

- I believe it performs the SSL certificate check for all domains, even HTTP ones (which will fail)
- Phpservermon has a related issue in which ports can't be changed after a server is entered (https://github.com/phpservermon/phpservermon/issues/418). Normally this isn't a problem and the server monitor works fine if the port is inaccurate, but this patch needs those ports to be accurate.

(Also, if someone is here trying to implement this for themselves, you'll need to modify the "servers" database table with two new fields for this patch to work, as described in the code of this PR; they don't automatically get created and the list of servers will "disappear" without an error message if those tables don't exist.)

Thanks for doing this!

TimZ99 committed:
Username validation fix (#512)

* Validation and (.) support

Added validation for the username during installation.
Added dot (.) to the list of allowed symbols for username.
Added ‘dots (.)’ to the error_user_name_invalid error message.

TimZ99 committed:
Fix for database populating bug (#502)

Fixed the population of the server table during installation.

acrylic-style committed:
Fixed spell missing and Translated new line (#535)

TimZ99 committed:
Added Japanese and updated project developers

TimZ99 committed:
Fixed #350 Email encoding

Removed utf8_decode().

Co-authored-by: shuhai <admin@4wei.cn>

TimZ99 committed:
Merge branch 'patch-2' of https://github.com/AndyGaskell/phpservermon into AndyGaskell-patch-2

TimZ99 committed:
Rename Nexmo.class.php to Nexmo.php (#436)

Co-authored-by: Andy Gaskell <andygaskelluk@yahoo.co.uk>

TimZ99 committed:
Added latest changes

TimZ99 committed:
Corrected Pushover comment

Changed sms to pushover.

TimZ99 committed:
Combined if-statements

Combined two of the same if-statements into one.

TimZ99 committed:
Clear log button - closes #383 (#505)

* Added button to clear the logs

Only admins can clear the logs. The content from the log and log_users table will
be deleted.

TimZ99 committed:
Added support for Telegram (#554)

Added support for Telegram. Notifications can be received in private chat or in groupchat.

This will close #389 and is a preview of what's coming next (#459) :wink:.

TimZ99 committed:
Updated version to 3.2.2

Version update needed to support Telegram.

TimZ99 committed:
Defined latency for pre-installed servers

Got deleted by accident in #502. 
Defined the latency for pre-installed servers on installation.

mtahiue committed:
Fixes #500 - updatePing (#516)

Moved rtime out of the if/else block, this fixes #500.

vahldiek committed:
run archive and cleanup per server (#515)

swiffer committed:
Fix method declaration error in PHP7.2 (#541)

Commit:
Fix #550 - Bug when sending FreeMobileSMS

TimZ99 committed:
Corrected indentation

TimZ99 committed:
Defined latency for new added servers

TimZ99 committed:
Added Telegram notifications

TimZ99 committed:
Added activation process for Telegram notifications

TimZ99 committed:
Added noopener to external link

Security update. More info can be found on https://developers.google.com/web/tools/lighthouse/audits/noopener.

TimZ99 committed:
Merge branch 'develop' into develop

TimZ99 committed:
Update Installer.php

TimZ99 commented:
The monitor marks the websites as offline when the SSL certificate is still valid, but just under the number of days, you specify. I would say that a warning is enough. Making the box yellow instead of red and still monitoring the latency could be a good solution.

TimZ99 commented on src/includes/functions.inc.php:
```diff
@@ -355,6 +355,21 @@ function psm_curl_get($href, $header = false, $body = true, $timeout = null, $ad
+ * @param string $href
+ * @return array
+ */
+function psm_cert_info_get($href, $port) {
```
Feedback in the conversation.

DominikWA commented:
Is it foreseeable when this will become part of the phpservermonitor? A web page without a valid certificate will appear to users as if it were unreachable, so the server monitor should report it.

TimZ99 commented:
@DominikWA  I think I will put it in v3.5.
